src/bootstrap/bin/rustc.rs

```diff
@@ -79,6 +79,17 @@ fn main() {
         cmd.args(lint_flags.split_whitespace());
     }
 
+    let crate_type = arg("--crate-type");
+
+    // When cross-compiling with LTO=thin/fat, Cargo invokes rustc
+    // with LTO enabled even for proc-macro crates, which causes compilation to fail.
+    // See https://github.com/rust-lang/rust/issues/110296 for more information.
+    //
+    // FIXME: maybe fix this on the cargo side?
```

Comment:
Are we sure this is Cargo's fault? It looks to me like we currently pass LTO flags directly (https://github.com/rust-lang/rust/blob/d5643b1dec7e97aa50e5545ee06e2fe6abeb3e4c/src/bootstrap/compile.rs#L871), rather than going through Cargo's profile settings.

I'm happy to merge with the comment updated to clarify this is not Cargo's profile options passing the LTO flags and (as such) the bug cannot be fixed in Cargo.

Reply:
> Are we sure this is Cargo's fault?

It seems like. There are a lot of extra arguments(such as `-Clink-args`, `-Cllvm-args`, `-Clto`) being passed into rustc invocations during cross-compilation.

If this code https://github.com/rust-lang/rust/blob/d5643b1dec7e97aa50e5545ee06e2fe6abeb3e4c/src/bootstrap/compile.rs#L871 is causing the problem, I think it should occur consistently, not only during cross-compilation.

Reply:
Outside of cross-compilation rustflags aren't passed to the host packages, I think. See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#target-applies-to-host, for some extra context (I think we don't set that option, but not sure).

Reply:
> Outside of cross-compilation rustflags aren't passed to the host packages, I think. See https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#target-applies-to-host, for some extra context (I think we don't set that option, but not sure).

Seems like this is the reason, thanks for the clarification.